docs/book/src/topics/managedcluster.md

```diff
@@ -277,6 +277,63 @@ User can do that by adding the Service Principal to the appropriate group define
 add the corresponding group ID in `spec.aadProfile.adminGroupObjectIDs`. 
 CAPI and CAPZ will be able to authenticate via AAD while accessing the target cluster.
 
+### Security Profile for AKS clusters.
+
+#### Workload Identity.
+Workloads deployed on an AKS cluster require Microsoft Entra application credentials or managed identities to access Microsoft Entra protected resources, such as Azure Key Vault and Microsoft Graph. Microsoft Entra Workload ID integrates with the capabilities native to Kubernetes to federate with external identity providers. Microsoft Entra Workload ID uses Service Account Token Volume Projection enabling pods to use a Kubernetes identity (that is, a service account). A Kubernetes token is issued and OIDC federation enables Kubernetes applications to access Azure resources securely with Microsoft Entra ID based on annotated service accounts.
```

Comment:
I'd vote to only link to https://learn.microsoft.com/azure/aks/workload-identity-overview instead of copy-pasting the content here. And instead of here, I think we should put the link in the Go doc comment so it gets generated into the rest of the API docs.

If there's anything else here that's specific to CAPZ that we can't reasonably include in the API documentation, then that makes sense to keep. Otherwise, I think we might not need most or all of these new docs here.